gnovm/stdlibs/stdshim/coins.gno

```diff
@@ -12,21 +15,62 @@ type Coin struct {
 	Amount int64  `json:"amount"`
 }
 
+// NewCoin returns a new coin with a denomination and amount
+func NewCoin(denom string, amount int64) Coin {
+	return Coin{
+		Denom:  denom,
+		Amount: amount,
+	}
+}
+
+// String provides a human-readable representation of a coin
 func (c Coin) String() string {
+	if c.IsZero() {
+		return ""
+	}
```

Comment:
Same comment as to relating `std` code

Reply:
stdshim was pending a question for @thehowl - now std & stdshim coins.gno are the same: 

[b423d29](https://github.com/gnolang/gno/pull/2104/commits/b423d2944c8600050ce7ad3e8e304caf288ae010)